File: string/1903_largest_odd_number_in_string/1903_largest_odd_number_in_string.go
package main

import "fmt"

func largestOddNumber(num string) string {
	n := len(num)
	for i := n - 1; i >= 0; i-- {
		if int(num[i])%2 == 1 {
			return num[0 : i+1]
		}
	}
	return ""

}

func main() {
	num := "52"
	res := largestOddNumber(num)
	fmt.Println(res)
}

File: string/1903_largest_odd_number_in_string/1903_largest_odd_number_in_string.py
class Solution:
    def largestOddNumber(self, num: str) -> str:
        n = len(num)
        for i in range(n-1, -1, -1):
            if int(num[i]) % 2 ==1:
                return num[0:i+1]
        return ""
        


num = "52"
res = Solution().largestOddNumber(num)
print(res)